src/debugbar/providers/DebugProvider.py

```diff
@@ -18,13 +23,22 @@ def register(self):
         debugger.add_collector(MessageCollector())
         debugger.add_collector(KeyValueCollector("Environment"))
         debugger.add_collector(time)
-        # debugger.add_collector(KeyValueCollector("Request"))
+        debugger.add_collector(KeyValueCollector("Request"))
         debugger.add_collector(QueryCollector().start_logging('masoniteorm.connection.queries'))
         self.application.bind('debugger', debugger)
 
     def boot(self):
         self.application.make('debugger').get_collector('Time').stop_measure('boot')
         response = self.application.make('response')
+        storage = self.application.make('storage')
         if 'text/html' in response.header('Content-Type'):
+            # Delete the contents of the directory first
+            files = glob.glob('storage/app/debug/*')
+            for f in files:
+                os.remove(f)
+
+            storage.disk('debug').put(f"x{str(time.time())+'-'+random_string(10)}.json", json.dumps(self.application.make('debugger').to_dict()))
             response.content += self.application.make('debugger').get_renderer('javascript').render()
             response.make_headers()
+        else:
+            storage.disk('debug').put(f"x{str(time.time())+'-'+random_string(10)}.json", json.dumps(self.application.make('debugger').to_dict()))
```

Comment:
@girardinsamuel Isn't this line equal to in 40th line? maybe deletion should be done inside if and others outside the flow?

Reply:
ahah well might be better. for now we are just experimenting with the system. we will try to refactor this cleanly at the end 😉